### PR TITLE
Add conversion for memref.alloc to _mlir_memref_to_llvm_alloc

### PR DESCRIPTION
**Pull request by JosseVanDelm**

### CODE REVIEW

File: kernels/alloc/main.c
Comment (jorendumoulin) [inline kernels/alloc/main.c:1]: This test does not actually check if the allocation is successful i think? It prints out the results, but does not check if they are equal. If the allocation\synchronisation fails, this goes unnoticed in CI. Maybe write some data with core 1 and then check the data with core 2 to determine if they have access to the same allocated buffer?

File: runtime/tomlir16.py [hunk 1]
```diff
@@ -18,8 +18,12 @@
 
     # swap attributes and region from place, and remove <> from attributes
     ir = re.sub(
-        r"<({.*?})>\s*(\({[^}]*}\))", lambda x: f"{x.group(2)} {x.group(1)}", ir
+        r"<({.*})>\s*(\({[\S\s]*<{.*}>[^}]*}\))",
+        lambda x: f"{x.group(2)} {x.group(1)}",
```
Comment (jorendumoulin) [inline runtime/tomlir16.py:22]: L̷̹̒͆͜ô̵̻̼͂o̴̼̠̐̇k̸͔̍ś̶̮̭͛ ̸̨̛̝̔v̴͖̥́̅e̴̗͝r̷̠͗̚y̶̞͝ ̵͈̓́g̴̢̬̈́͠o̵̤͆o̸̦̭͠d̶͍͙̽!̴̓̀͜!̶̻͍͐

File: kernels/alloc/main.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+int8_t *allocated_pointer;
+
+int8_t *_mlir_memref_to_llvm_alloc(uint32_t size) {
+  /* This calls malloc on the DMA core
+   * --> requires mlir opt to compile with:
+   *  --convert-memref-to-llvm="use-generic-functions index-bitwidth=32"
+   * To ensure that all cores in the cluster come up with the correct
+   */
+  snrt_cluster_hw_barrier();
```
Comment (jorendumoulin) [inline kernels/alloc/main.c:17]: This cluster synchronization is unnecessary?

**Review by jorendumoulin — CHANGES_REQUESTED** [review]
Very nice! However, the test does not actually test the correctness of the allocation, I think.

File: kernels/alloc/main.c
Comment (JosseVanDelm) [reply kernels/alloc/main.c:1]: Good call :smile:

**Review by jorendumoulin — APPROVED** [review]
Perfect! :star_struck:

**Pull request merged**